src/transforms/ansi_stripper.rs

```diff
@@ -0,0 +1,127 @@
+use super::Transform;
+use crate::{
+    event::{self, ValueKind},
+    topology::config::{DataType, TransformConfig, TransformDescription},
+    Event,
+};
+use serde::{Deserialize, Serialize};
+use string_cache::DefaultAtom as Atom;
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct AnsiStripperConfig {
+    field: Option<Atom>,
+}
+
+inventory::submit! {
+    TransformDescription::new_without_default::<AnsiStripperConfig>("ansi_stripper")
+}
+
+#[typetag::serde(name = "ansi_stripper")]
+impl TransformConfig for AnsiStripperConfig {
+    fn build(&self) -> crate::Result<Box<dyn Transform>> {
+        let field = self.field.as_ref().unwrap_or(&event::MESSAGE);
+
+        Ok(Box::new(AnsiStripper {
+            field: field.clone(),
+        }))
+    }
+
+    fn input_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn output_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn transform_type(&self) -> &'static str {
+        "ansi_stripper"
+    }
+}
+
+pub struct AnsiStripper {
+    field: Atom,
+}
+
+impl Transform for AnsiStripper {
+    fn transform(&mut self, mut event: Event) -> Option<Event> {
+        let log = event.as_mut_log();
+
+        match log.get_mut(&self.field) {
+            None => debug!(
+                message = "Field does not exist.",
+                field = self.field.as_ref(),
+            ),
+            Some(ValueKind::Bytes(ref mut bytes)) => {
+                *bytes = match strip_ansi_escapes::strip(bytes.clone()) {
```

Comment:
A downside to this is that the clone happens for all strings, even those that don't have any escape sequences.

One (initial) solution could be to match the bytes for the `\x1b[` pattern and only _then_ strip escape sequences, but that would only cover a subset of possible escape sequences.

Reply:
I think this is a good place to start. Down the line, if there's a need for more performance, we can make a pass at profiling and optimizing.